### ISSUE TITLE
SPSearchFileType Key Issue

### ISSUE THREAD

**Issue by mrpullen** [PowerShell/SharePointDsc#542]
**Details of the scenario you try and problem that is occurring:**
Attempting to add SPSearchFileType to multiple Search Service Applications inside of a single configuration.

**The DSC configuration that is using the resource:**
```
foreach($searchFileType in $SearchFileTypes) {
		SPSearchFileType "Portal$($searchFileType.FileType)"
		{
			FileType = "$($searchFileType.FileType)"
			ServiceAppName = "Search Service"
			Description = "$($searchFileType.Description)"
			MimeType = "$($searchFileType.MimeType)"
			Ensure = "Present"
			InstallAccount = $installAccount
		}
	}

foreach($searchFileType in $SearchFileTypes) {
		SPSearchFileType "App$($searchFileType.FileType)"
		{
			FileType = "$($searchFileType.FileType)"
			ServiceAppName = "App Search Service"
			Description = "$($searchFileType.Description)"
			MimeType = "$($searchFileType.MimeType)"
			Ensure = "Present"
			InstallAccount = $installAccount
		}
}
```

**Version of the Operating System and PowerShell the DSC Target Node is running:**
WIN2008R2
**Version of the DSC module you're using:**
V 1.5

ERROR: 

Add-NodeKeys : The key properties combination 'pdf' is duplicated for keys 'FileType' of resource 
'SPSearchFileType' in node ##NODENAME##. Please make sure key properties are unique for each resource in a node.

**Fix:**
Add the ServiceAppName as a Key on this resource as well.

**Comment by BrianFarnhill** [comment]
Agreed this is a bug, the service app name should have also been marked as a key. This should be a quick fix for us to put in though. Thanks for reporting.

**Comment by BrianFarnhill** [comment]
Fixed in dev, this will release in 1.7 - closing this issue.
